helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -850,6 +858,56 @@ public static JobConfig getJobConfig(HelixManager manager, String job) {
     return workflowConfigMap;
   }
 
+  /**
+   * Batch get the configurations of all workflows in this cluster
+   * within the specified timeout in milliseconds.
+   *
+   * @param timeout a long integer presents the timeout, in milliseconds
+   * @return a map of <String, WorkflowConfig>
+   * @throws InterruptedException if the future thread was interrupted
+   * @throws ExecutionException if the future task completed exceptionally
+   * @throws TimeoutException if waiting for result timed out
+   */
+  public Map<String, WorkflowConfig> getWorkflows(long timeout)
+    throws CancellationException, InterruptedException, ExecutionException, TimeoutException {
+    if (timeout <= 0L) {
+      throw new IllegalArgumentException("timeout must be positive.");
+    }
+    
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
Create a new executor on every call is not optimal. Can we have a static object?

Reply:
When I was implementing this, I also thought about it. I don't know if this call is heavy or not. If the call is rare, I think it is fine to create a new executor. If it is heavy or called frequently, we definitely want to optimize it and keep the single thread running.

Reply:
Do you have an idea of how frequently the api is called? To improve performance, is it fine to cache the results?

Reply:
It is not related to how frequently it is called. And in reality, we don't know. It depends on our customer application's logic. We need to reasonably optimize our code to support the possible use cases.

In addition, for caching the result, you need to think about how to refresh the cache when things changed. Probably be too complicated here.

Reply:
Keeping a static executor is doable. It means we create an executor with a threadpoolSize pool. And need to have a good size for the threadpool based on the requests. If it is too small, tasks may be blocked in the wait queue for submission. Do you agree?

Reply:
The requests mostly likely won't come in parallel. Don't need to over-engineering the problem, the SLA here is each request sets a time_out, whether it doesn't finish due to pool size too small or the task execution too long.